modules/ebs_snapshot_explorer/main.py

```diff
@@ -0,0 +1,303 @@
+    Returns:
+        dict: Mapping regions to corresponding list of volume_ids.
+    """
+    ec2_data = deepcopy(session.EC2)
+    if 'Volumes' not in ec2_data:
+        fields = ['EC2', 'Volumes']
+        module = 'enum_ebs_volumes_snapshots'
+        fetched_ec2_instances = pacu.fetch_data(fields, module)
+        if fetched_ec2_instances is False:
+            return None
+    volume_ids = []
+    for volume in ec2_data['Volumes']:
+        if volume['Region'] == region:
+            volume_ids.append(volume['VolumeId'])
+    return volume_ids
+
+
+def generate_volumes_from_snapshots(client, snapshots, zone):
+    """Returns a list of generated volumes"""
+    volume_ids = []
+    waiter = client.get_waiter('snapshot_completed')
+    waiter.wait(SnapshotIds=snapshots)
+    for snapshot in snapshots:
+        response = client.create_volume(
+            SnapshotId=snapshot, AvailabilityZone=zone)
+        volume_ids.append(response['VolumeId'])
+    store_temp_data({'volumes':volume_ids})
+    return volume_ids
+
+
+def generate_snapshots_from_volumes(client, volume_ids):
+    """Returns a list of generated snapshots from volumes"""
+    snapshot_ids = []
+    for volume in volume_ids:
+        response = client.create_snapshot(VolumeId=volume)
+        snapshot_ids.append(response['SnapshotId'])
+
+    store_temp_data({'snapshots': snapshot_ids})
+    return snapshot_ids
+
+
+def delete_volumes(client, volumes):
+    """Deletes a given list of volumes
+
+    If the volume is in use, the volume is forcibly detached because this module
+    only deals with temporary copies so data integrity is not a high priority when
+    a volume is ready to be detatched. After the volume is forcibly detatched, the
+    volume will be deleted after the detaching operation finishes.
+    """
+    failed_volumes = []
+    for volume in volumes:
+        try:
+            client.delete_volume(VolumeId=volume)
+        except ClientError as error:
+            code = error.response['Error']['Code']
+            if code == 'VolumeInUse':
+                client.detach_volume(
+                    VolumeId=volume,
+                    Force=True)
+                waiter = client.get_waiter('volume_available')
+                waiter.wait(VolumeIds=[volume])
+                client.delete_volume(VolumeId=volume)
+                continue
+            failed_volumes.append(volume)
+    return failed_volumes
+
+
+def delete_snapshots(client, snapshots):
+    """Deletes a given list of snapshots"""
+    failed_snapshots = []
+    for snapshot in snapshots:
+        try:
+            client.delete_snapshot(SnapshotId=snapshot)
+        except ClientError:
+            failed_snapshots.append(snapshot)
+    return failed_snapshots
+
+
+def cleanup(client):
+    """Cleans up the temporary snapshots and volumes created during this
+    modules execution
+    """
+    new_data = {}
+    success = True
+    temp_file = Path(__file__).parent / 'temp.json'
```

Comment:
Tested this and it seems to work, but seems like it should be trying to divide with `/` instead of concatting just by the looks of it

Reply:
I'm not sure how it works, it might be operator overload for Path objects, but in the official docs, it looks like that's the proper way to append parts to a Path object.